app/license.go

```diff
@@ -74,10 +74,10 @@ func (s *Server) SaveLicense(licenseBytes []byte) (*model.License, *model.AppErr
 	record.Id = license.Id
 	record.Bytes = string(licenseBytes)
 
-	_, err = s.Store.License().Save(record)
-	if err != nil {
+	_, nErr := s.Store.License().Save(record)
+	if nErr != nil {
 		s.RemoveLicense()
-		return nil, model.NewAppError("addLicense", "api.license.add_license.save.app_error", nil, "err="+err.Error(), http.StatusInternalServerError)
+		return nil, model.NewAppError("addLicense", "api.license.add_license.save.app_error", nil, err.Error(), http.StatusInternalServerError)
```

Comment:
`Save` also returns AppError. Don't we need to check that here and return the apperror directly?